pkg/controller/unifiedpushserver/unifiedpushserver_controller.go

```diff
@@ -736,3 +739,63 @@ func containsCronJob(cronJobs []batchv1beta1.CronJob, candidate *batchv1beta1.Cr
 	}
 	return false
 }
+
+func applySpecDefaultValues(instance *pushv1alpha1.UnifiedPushServer) {
+
+	if instance.Spec.UnifiedPushMemoryLimit == "" {
+		instance.Spec.UnifiedPushMemoryLimit = cfg.UPSMemoryLimit
```

Comment:
Does this actually set the value in the CR spec? Is it a good idea for operators to update spec fields in CRs? :thinking:

Reply:
similarly, I just took the example from our other operators.

Reply:
Can you share a link to those?

Reply:
I'd recommend do it as it was done here[1]. Create a config which will set the values when they are not defined.  

[1] - https://github.com/aerogear/mobile-security-service-operator/blob/master/pkg/controller/mobilesecurityservice/mandatory_specs.go

Reply:
https://github.com/aerogear/mobile-security-service-operator/blob/master/pkg/controller/mobilesecurityservice/controller.go#L175

There's 1 only. I thought there was more than one.

Reply:
In MSS Oper we have a default setup for each type in order to not hurt concepts as cohesion and encapsulation.